android/src/org/ppsspp/ppsspp/NativeActivity.java

```diff
@@ -381,8 +353,62 @@ public void onCreate(Bundle savedInstanceState) {
 				setupSystemUiCallback();
 			}
 		}
+    	updateDisplayMetrics(null);
+
+		NativeApp.computeDesiredBackbufferDimensions();
+		int bbW = NativeApp.getDesiredBackbufferWidth();
+		int bbH = NativeApp.getDesiredBackbufferHeight();
+
+        mSurfaceView = new NativeSurfaceView(NativeActivity.this, bbW, bbH);
+        mSurfaceView.getHolder().addCallback(NativeActivity.this);
+        Log.i(TAG, "setcontentview before");
+		setContentView(mSurfaceView);
+        Log.i(TAG, "setcontentview after");
+
+		mRenderLoopThread = new Thread(mEmulationRunner);
+		mRenderLoopThread.start();
     }
 
+	@Override
+	public void surfaceCreated(SurfaceHolder holder)
+	{
+		Log.d(TAG, "Surface created.");
+	}
+
+	@Override
+	public void surfaceChanged(SurfaceHolder holder, int format, int width, int height)
+	{
+		Log.w(TAG, "Surface changed. Resolution: " + width + "x" + height + " Format: " + format);
+		// Make sure we have fresh display metrics so the computations go right.
+		// This is needed on some very old devices, I guess event order is different or something...
+		Point sz = new Point();
+        updateDisplayMetrics(sz);
+        NativeApp.backbufferResize(width, height, format);
```

Comment:
Yeah, I'll have see if I can configure Eclipse better.. for some reason this always turns into a mess.